### PR TITLE
Add project skeleton for Azure DigitalTwins Model Parser

### PR DESCRIPTION
**Pull request by azabbasi**
Initial commit to check in the project skeleton and client builder structure. 
The namespace is a placeholder and is subject to change.
The service directory name is also subject to change.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -48,6 +48,9 @@
 # PRLabel: %Digital Twins
 /sdk/digitaltwins/                                  @drwill-ms @timtay-microsoft @abhipsaMisra @vinagesh @azabbasi @bikamani @barustum
 
+# PRLabel: %DigitalTwins Parser
```
Comment (drwill-ms) [inline .github/CODEOWNERS:51]: missing james

File: pom.xml [hunk 2]
```diff
@@ -78,6 +78,9 @@
         <!-- Mixed Reality -->
         <module>sdk/mixedreality/azure-mixedreality-authentication</module>
 
+        <!-- DigitalTwins Model Parser -->
+        <module>sdk/modelparser/azure-digitaltwins-parser</module>
```
Comment (drwill-ms) [inline pom.xml:82]: digital twins parser sounds like a thing that parses digital twins, not digital twins models. I know you aren't defining the name, rather going with the precedent from other languages, but I think it is a worthwhile question to bring up. Thoughts?

Comment (azabbasi) [reply pom.xml:82]: I absolutely agree, that's why I left it for debate, I think specifying the `model` is necessary here.

Comment (azabbasi) [reply pom.xml:82]: I will start a conversation about the namespace to the team and the PMs.

File: sdk/modelparser/azure-digitaltwins-parser/README.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Next steps
+
+- TBD
+
+## Contributing
+
+This project welcomes contributions and suggestions.
+Most contributions require you to agree to a Contributor License Agreement (CLA) declaring that you have the right to, and actually do, grant us the rights to use your contribution.
+For details, visit <https://cla.microsoft.com.>.
```
Comment (drwill-ms) [inline sdk/modelparser/azure-digitaltwins-parser/README.md:54]: no period in the URL within the angle braces

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/ParserServiceVersion.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+package com.azure.digitaltwins.parser;
+
+import com.azure.core.util.ServiceVersion;
+
+/**
+ * The service API versions of Azure DigitalTwins Model Parser that are supported by this client.
+ */
+public enum ParserServiceVersion implements ServiceVersion {
+    V2021_02_11("2021_02_11");
```
Comment (drwill-ms) [inline sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/ParserServiceVersion.java:12]: Is this a bogus API version just to meet track2 guidelines?

What value is used in the other languages?

File: sdk/modelparser/azure-digitaltwins-parser/src/samples/README.md [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# IoT Models Repository Samples
```
Comment (drwill-ms) [inline sdk/modelparser/azure-digitaltwins-parser/src/samples/README.md:1]: Not IoT right?

File: sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/ParserServiceVersion.java (hunk 4 above)
Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/main/java/com/azure/digitaltwins/parser/ParserServiceVersion.java:12]: we are the pioneers at the moment. There is no API call at all so I think we can get away with not having any API versions at all.

File: sdk/modelparser/azure-digitaltwins-parser/src/samples/README.md (hunk 5 above)
Comment (drwill-ms) [reply sdk/modelparser/azure-digitaltwins-parser/src/samples/README.md:1]: I think this is just copy pasta error

Comment (azabbasi) [reply sdk/modelparser/azure-digitaltwins-parser/src/samples/README.md:1]: Not Models Repository either ... fixing

**Review by drwill-ms — APPROVED** [review]

**Review by barustum — APPROVED** [review]

**Pull request merged**